apps/src/applab/dataBrowser/EditTableRow.jsx

```diff
@@ -1,35 +1,100 @@
+import FirebaseStorage from '../firebaseStorage';
 import Radium from 'radium';
 import React from 'react';
+import { castValue, displayValue } from './dataUtils';
 import * as dataStyles from './dataStyles';
 
 const EditTableRow = React.createClass({
   propTypes: {
     columnNames: React.PropTypes.array.isRequired,
+    tableName: React.PropTypes.string.isRequired,
     record: React.PropTypes.object.isRequired
   },
 
+  getInitialState() {
+    return {
+      isEditing: false
+    };
+  },
+
+  handleChange(columnName, event) {
+    const newRecord = Object.assign({}, this.state.newRecord, {
+      [columnName]: castValue(event.target.value)
+    });
+    this.setState({ newRecord });
+  },
+
+  handleSave() {
+    FirebaseStorage.updateRecord(
+      this.props.tableName,
+      this.state.newRecord,
+      () => this.setState({ isEditing: false }),
+      this.handleError
+    );
+  },
+
+  handleEdit() {
+    this.setState({
+      isEditing: true,
+      newRecord: this.props.record
+    });
+  },
+
+  handleError(msg) {
+    console.warn(msg);
+  },
```

Comment:
Here you chose to create a handleError method. Elsewhere, you used an inline arrow method. Would be good to be consistent.